### ISSUE TITLE
HelmRelease created by Kustomization won't reconcile changes when manualy edited

### ISSUE THREAD

**Issue by mmckane** [fluxcd/flux2#2025]
### Describe the bug

HelmReleases installed via Kustomization resources can be modified in cluster via kubectl and Flux will not remove the changes to match what is in git when reconcilliation is triggered.

### Steps to reproduce

1. Create a Flux V2 HelmRelease spec and apply it to the server with a Flux V2 Kustomization spec
2. Reconcile Kustomization with Flux this will apply the HelmRelease to the cluster and install it.
3. Edit the Helmrelease settings manually via Kubectl edit helmrelease <name of release> -n <namespace> for example add
```yaml
spec:
  update:
     force: true
    remediation:
      retries: -1
```
4. Reconcile Kustomization via flux reconcile kustomization <name of kustomization>. Reconcile succeeds without errors
5. kubectl get helmRelease <name> -n <namespace> -o yaml. The spec above will stay the same and not be removed to match the code in git.

### Expected behavior

The HelmRelease is replaced with what is in git.

Would note that it's ok that if helmReleases are only patched, vs replaced with what is in git if they are manually edited. but would argue this should be called out in the documentation somewhere if this is the case.

### OS / Distro

Ubuntu 20.04

### Flux version

18.3

### Flux check

► checking prerequisites
✔ Kubernetes 1.21.4-gke.200 >=1.19.0-0
► checking controllers
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v0.12.0
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v0.15.5
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v0.17.1
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v0.16.0
✔ all checks passed

### Git provider

Github Enterprise

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by stefanprodan** [comment]
Can you please upgrade your cluster to flux2 0.21.0 and test this one more time?

**Comment by mmckane** [comment]
@stefanprodan updated cluster to 0.21.0 tested the changes to the spec. also tried just adding a label to see if it would be removed. Label remained kind: helmRelease still appears to be doing some kind of merge with manual edits instead of being updated/replaced with what is in git.

**Comment by stefanprodan** [comment]
This was fixed in Flux 2.0 GA
